nav2_behavior_tree/include/nav2_behavior_tree/bt_service_node.hpp

```diff
@@ -128,10 +128,17 @@ class BtServiceNode : public BT::ActionNodeBase
   BT::NodeStatus tick() override
   {
     if (!request_sent_) {
+      // reset the flag to send the request or not, allowing the user the option to set it in on_tick
+      should_send_request_ = true;
+
+      // user defined callback, may modify "should_send_request_".
       on_tick();
-      future_result_ = service_client_->async_send_request(request_).share();
-      sent_time_ = node_->now();
-      request_sent_ = true;
+
+      if (should_send_request_) {
+        future_result_ = service_client_->async_send_request(request_).share();
+        sent_time_ = node_->now();
+        request_sent_ = true;
+      }
     }
     return check_future();
```

Comment:
I have a small doubt what to do here. if `should_send_request_` then `future_result_` is null and we can't `check_future`. What should we do in this case? Should the Node return Failure?

Reply:
I'd log that the node set that it was not valid to send request and then return `BT::NodeStatus::FAILURE`